test/Microsoft.AspNet.Mvc.Core.Test/ViewDataDictionaryOfTModelTest.cs

```diff
@@ -193,11 +193,9 @@ public void CopyConstructors_OverrideSourceMetadata_IfDeclaredTypeChanged()
             // Assert
             Assert.NotNull(viewData1.ModelMetadata);
             Assert.Equal(expectedType, viewData1.ModelMetadata.ModelType);
-            Assert.Equal(expectedType, viewData1.ModelMetadata.RealModelType);
 
             Assert.NotNull(viewData2.ModelMetadata);
             Assert.Equal(expectedType, viewData2.ModelMetadata.ModelType);
-            Assert.Equal(expectedType, viewData2.ModelMetadata.RealModelType);
```

Comment:
nit: suggest instead adding `Assert.NotNull(viewDataN.ModelExplorer);` and `Assert.Equal(expectedType, viewDataN.ModelExplorer.ModelType);` assertions